### PR TITLE
[AMP OP&Test] adjust test_elementwise_sub's tolerance, max_relative_error of grad and

### PR DESCRIPTION
**Pull request by Vvsmile**
### PR types

Others
### PR changes

Others
### Describe

Adjust the fp16 tolerance, max_relative_error of grad and atol/rtol of output to 1e-3

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py [hunk 1]
```diff
@@ -183,35 +371,122 @@ def setUp(self):
...
+        self.dtype = np.float16
+
+
+@unittest.skipIf(
+    not core.is_compiled_with_cuda()
+    or not core.is_bfloat16_supported(core.CUDAPlace(0)),
+    "core is not compiled with CUDA and do not support bfloat16",
+)
+class TestElementwiseBF16OP_scalar(TestElementwiseOp):
+    def setUp(self):
```
Comment (ZzSean) [inline python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py]: 这种scalar的是不是不用测反向

File: python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py [hunk 2]
```diff
@@ -69,15 +74,53 @@ def if_enable_cinn(self):
...
+        }
+        self.outputs = {'Out': self.inputs['X'] - self.inputs['Y']}
+        self.inputs = {
+            'X': convert_float_to_uint16(self.inputs['X']),
+            'Y': convert_float_to_uint16(self.inputs['Y']),
+        }
+        self.outputs = {'Out': convert_float_to_uint16(self.outputs['Out'])}
+
+    def test_check_grad_normal(self):
+        self.check_grad(['X', 'Y'], 'Out')
```
Comment (ZzSean) [inline python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py]: 需要用`check_grad_with_place`,否则跑到CPU会出错，下同

File: python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py [hunk 3]
```diff
@@ -90,15 +133,62 @@ def if_enable_cinn(self):
         self.enable_cinn = False
 
 
+class TestElementwiseSubFP16OP_ZeroDim1(TestElementwiseOp):
```
Comment (ZzSean) [inline python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py]: 继承的父类不对吧，下同

File: python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py [hunk 4]
```diff
@@ -90,15 +133,62 @@ def if_enable_cinn(self):
...
+    def init_dtype(self):
+        self.dtype = np.float16
+
+
+@unittest.skipIf(
+    not core.is_compiled_with_cuda()
+    or not core.is_bfloat16_supported(core.CUDAPlace(0)),
+    "core is not compiled with CUDA and do not support bfloat16",
+)
+class TestElementwiseBF16OP_ZeroDim1(TestElementwiseOp):
```
Comment (ZzSean) [inline python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py]: 直接继承`TestElementwiseBF16OP`，下同

File: python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py [hunk 5]
```diff
@@ -69,15 +74,53 @@ def if_enable_cinn(self):
...
+        self.inputs = {
+            'X': np.random.uniform(0.1, 1, [2, 3, 4, 5]).astype(np.float32),
+            'Y': np.random.uniform(0.1, 1, [2, 3, 4, 5]).astype(np.float32),
+        }
+        self.outputs = {'Out': self.inputs['X'] - self.inputs['Y']}
+        self.inputs = {
+            'X': convert_float_to_uint16(self.inputs['X']),
+            'Y': convert_float_to_uint16(self.inputs['Y']),
+        }
+        self.outputs = {'Out': convert_float_to_uint16(self.outputs['Out'])}
```
Comment (ZzSean) [inline python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py:102]: 少了self.if_check_prim() self.if_enable_cinn()

Comment (Vvsmile) [reply python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py:102]: Done

File: python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py [hunk 6]
```diff
@@ -66,7 +71,49 @@ def if_check_prim(self):
         self.check_prim = True
 
     def if_enable_cinn(self):
-        pass
+        self.enable_cii = False
```
Comment (ZzSean) [inline python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py]: 拼写错误

**Review by ZzSean — APPROVED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**